ntb_hw_switchtec.c

```diff
@@ -125,6 +125,8 @@ struct switchtec_ntb {
 	enum ntb_width link_width;
 	struct work_struct check_link_status_work;
 	bool link_force_down;
+
+	struct kobject kobj;
```

Comment:
This violates the documentation:
>   No structure should **EVER** have more than one kobject embedded within it.
>  If it does, the reference counting for the object is sure to be messed
>  up and incorrect, and your code will be buggy.  So do not do this. [[kobject]]

switchtec_ntb has a kobj already in ntb.dev.kobj.

[kobject]: https://elixir.bootlin.com/linux/latest/source/Documentation/core-api/kobject.rst

Reply:
Will update. Thanks!